worksheet_D/oxo.py

```diff
@@ -0,0 +1,222 @@
+
+            # Bottom line
+        elif x == 0 and y == 2:
+            return board.fields["Bottom Left"]
+        elif x == 1 and y == 2:
+            return board.fields["Bottom Middle"]
+        elif x == 2 and y == 2:
+            return board.fields["Bottom Right"]
+
+    def set_square(self, x, y, mark):
+        """
+        For nxn I would again have to add extra squares
+        """
+
+            # Top line
+        if x == 0 and y == 0 and board.fields["Top Left"] == 0:
+                # Change the field value to mark
+            board.fields["Top Left"] = mark
+            return True
+        elif x == 0 and y == 0 and board.fields["Top Left"] > 0:
+                # if the field is anything other than 0 return False
+            return False
+        if x == 1 and y == 0 and board.fields["Top Middle"] == 0:
+            board.fields["Top Middle"] = mark
+            return True
+        elif x == 1 and y == 0 and board.fields["Top Middle"] > 0:
+            return False
+        if x == 2 and y == 0 and board.fields["Top Right"] == 0:
+            board.fields["Top Right"] = mark
+            return True
+        elif x == 2 and y == 0 and board.fields["Top Right"] > 0:
+            return False
+
+            # Middle line
+        if x == 0 and y == 1 and board.fields["Middle Left"] == 0:
+            board.fields["Middle Left"] = mark
+            return True
+        elif x == 0 and y == 1 and board.fields["Middle Left"] > 0:
+            return False
+        if x == 1 and y == 1 and board.fields["Middle Middle"] == 0:
+            board.fields["Middle Middle"] = mark
+            return True
+        elif x == 1 and y == 1 and board.fields["Middle Middle"] > 0:
+            return False
+        if x == 2 and y == 1 and board.fields["Middle Right"] == 0:
+            board.fields["Middle Right"] = mark
+            return True
+        elif x == 2 and y == 1 and board.fields["Middle Right"] > 0:
+            return False
+
+            # Bottom line
+        if x == 0 and y == 2 and board.fields["Bottom Left"] == 0:
+            board.fields["Bottom Left"] = mark
+            return True
+        elif x == 0 and y == 2 and board.fields["Bottom Left"] > 0:
+            return False
+        if x == 1 and y == 2 and board.fields["Bottom Middle"] == 0:
+            board.fields["Bottom Middle"] = mark
+            return True
+        elif x == 1 and y == 2 and board.fields["Bottom Middle"] > 0:
+            return False
+        if x == 2 and y == 2 and board.fields["Bottom Right"] == 0:
+            board.fields["Bottom Right"] = mark
+            return True
+        elif x == 2 and y == 2 and board.fields["Bottom Right"] > 0:
+            return False
+
+    def is_board_full(self):
+        """
+        Nothing would change for nxn
+        """
+
+            # Loop through the dictionary and if any of the values is 0 return False
+        for field_name, value in board.fields.iteritems():
```

Comment:
This is OK. Note that if you only want the values, you can use `itervalues` instead of `iteritems`